subsys/bluetooth/common/Kconfig

```diff
@@ -94,20 +94,37 @@ config BT_BUF_ACL_RX_COUNT
 	  When Controller to Host flow control is not enabled the Controller
 	  can assume that the Host has infinite amount of buffers.
 
+config BT_BUF_CMD_COMPLETE_EVT_SIZE
+	  int "Maximum supported HCI Command Complete Event buffer length"
+	  # LE Read Supported Commands command complete event.
+	  default 68
+	  range 68 255
+	  help
+	    Maximum supported HCI Command Complete event buffer size.
+	    This value does not include the HCI Event header. This value is
+	    used by both the Host and the Controller for buffer sizes that
+	    include HCI Command Complete events. It should be set according
+	    to the expected HCI Command Complete events that can be generated
+	    from the configuration.
+
+	    If the subset of possible HCI Command Complete events is unknown,
+	    this should be set to the maximum of 255.
+
 config BT_BUF_EVT_RX_SIZE
 	int "Maximum supported HCI Event buffer length"
 	default 255 if (BT_EXT_ADV && BT_OBSERVER) || BT_PER_ADV_SYNC || BT_DF_CONNECTION_CTE_RX || BT_CLASSIC
-	# LE Read Supported Commands command complete event.
-	default 68
-	range 68 255
+	default BT_BUF_CMD_COMPLETE_EVT_SIZE
+	range BT_BUF_CMD_COMPLETE_EVT_SIZE 255
```

Comment:
Is the range and default value here due to the fallback mechanism you have defined? 

If we did not fallback to the event buffers in case of error, could this be reduced? Do we have any events larger than ~32 without ext_adv?

Reply:
I have also thought of a similar approach, but considering that almost all configurations now use extended advertising as a default option, the value of this optimization is not high.

Reply:
The current design will add about 140 bytes of RAM overhead, but it ensures relative safety, so I think it is worth it and will not be noticeable to users.